### ISSUE TITLE
Invalid datetime format database error when posting emoji

### ISSUE THREAD

**Issue by 9662** [nextcloud/social#293]
While testing this application (and aware that federation with Mastodon hosts may not be working), I saw the following error in the Nextcloud log:

```
[no app in context] Warning: 500 - {"status":-1,"exception":"Doctrine\\DBAL\\Exception\\DriverException","message":"An exception occurred while executing 'INSERT INTO `oc_social_server_notes` (`id`, `type`, `to`, `to_array`, `cc`, `bcc`, `content`, `summary`, `published`, `published_time`, `attributed_to`, `in_reply_to`, `source`, `instances`, `local`, `creation`) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [\"https:\\\/\\\/example.net\\\/apps\\\/social\\\/@user\\\/15464479942021935112\", \"Note\", \"\", \"[\\\"https:\\\/\\\/mastodon.xyz\\\/users\\\/remoteuser\\\"]\", \"[]\", \"[]\", \"@remoteuser@mastodon.xyz Hello there! \\ud83d\\ude00\", \"\", \"2019-01-02T16:53:14+00:00\", \"2019-01-02 16:53:14\", \"https:\\\/\\\/example.net\\\/apps\\\/social\\\/@user\", \"\", \"\", \"[{\\\"uri\\\":\\\"https:\\\/\\\/mastodon.xyz\\\/users\\\/remoteuser\\\/inbox\\\",\\\"type\\\":1,\\\"priority\\\":3}]\", \"1\", \"2019-01-02 16:53:16\"]:\n\nSQLSTATE[22007]: Invalid datetime format: 1366 Incorrect string value: '\\xF0\\x9F\\x98\\x80' for column 'content' at row 1"}

POST /apps/social/api/v1/post
from 0.0.0.0 by user at 2019-01-02T16:53:16+00:00
```
(I have anonymised some values)

I am not aware of the significance of this warning, but I thought I would mention it anyway. Apologies for lack of a more detailed report.

**Comment by daita** [comment]
I would require more details on the setup/type of your database, also can you paste the structure of the table `server_notes`: `SHOW COLUMNS FROM oc_social_server_notes;` 

Do you have the issue each time you try to make a new direct post ? 
What about a public post while you mention the @remoteuser@mastodon.xyz in the content of the post ?

Also, what happens when you send a toot from Mastodon to your instance of Social ?

**Comment by 9662** [comment]
Indeed, I followed the instruction in https://docs.nextcloud.com/server/stable/admin_manual/configuration_database/mysql_4byte_support.html (mistakenly, I have followed the instructions for MySQL and not for MariaDB, although that's what my distro comes with, but it seemed to work regardless) and now the error has disappeared from the logs **and** messages with emoji are correctly transferred between my Nextcloud instance and the Mastodon server. Awesome!

I am leaving this open for devs to check before closing. I still don't know why it complains about an "invalid datetime format".
